src/main/scala/com/microsoft/ml/spark/cognitive/AzureSearch.scala

```diff
@@ -326,7 +326,7 @@ object AzureSearchWriter extends IndexParser with SLogging {
   }
 
   def write(df: DataFrame, options: Map[String, String] = Map()): Unit = {
-    prepareDF(df, options).foreachPartition(it => it.foreach(_ => ()))
```

Comment:
foreachPartition == foreach? surprising.

Reply:
i think this is a ne3w func in spark3

Reply:
oh, so this only builds in spark3 now?

Reply:
yes as you can see there is really no way to do both 2.4 and 3. Best best for 2.4 is to use latest release and we can make hotfixes on there too